### PR TITLE
fix: botprojects context menu issues

### LINKED ISSUE #4811: Popup tooltip text truncation
Screen Shot 2020-11-13 at 12 01 31 PM

  
 

Screen Shot 2020-11-13 at 12 01 35 PM

All tooltips are being truncated here. After speaking to @sangwoohaan it looks like the text needs to wrap to the next line rather than getting truncated.

### LINKED ISSUE #4854: Invalid warning and errors notifications showing up
Screen Shot 2020-11-17 at 10 46 45 AM

Screen Shot 2020-11-17 at 11 11 22 AM

1. With the new skills workflow if the "Composer Local" endpoint is chosen we dont need to check for an empty endpoint url in the settings.
2. Missing skill manifest should not be a warning shown on the Root Bot
3. A warning for missing MsAppID is shown. This warning is not required. An MSAppID or AppPassword is no longer required to call skills

### PR DESCRIPTION
**Pull request by zhixzhan**
## Description
1. Fix invalid warning and errors show up in project tree
2. Fix ProjectTree Popup tooltip text truncation
3. Fix Links in context menu dont work.

## Task Item
close #4811
refs #4853
close #4854

## Screenshots
image

### CODE REVIEW

File: Composer/packages/lib/indexers/src/dialogIndexer.ts [hunk 1]
```diff
@@ -168,7 +174,8 @@ function extractReferredSkills(dialog): string[] {
   const visitor: VisitorFunc = (path: string, value: any): boolean => {
     // it's a valid schema dialog node.
     if (has(value, '$kind') && value.$kind === SDKKinds.BeginSkill) {
-      skills.push(value.skillEndpoint);
+      const skillName = getSkillNameFromSetting(value.skillEndpoint);
```
Comment (srinaath) [inline Composer/packages/lib/indexers/src/dialogIndexer.ts:177]: Nice!

File: Composer/packages/client/src/pages/design/DesignPage.tsx [hunk 2]
```diff
@@ -886,11 +886,10 @@ const DesignPage: React.FC<RouteComponentProps<{ dialogId: string; projectId: st
         {dialogId && (
           <CreateQnAModal dialogId={dialogId} projectId={projectId} qnaFiles={qnaFiles} onSubmit={handleCreateQnA} />
         )}
-        {displaySkillManifest && (
+        {displaySkillManifestOnProjectId && (
           <DisplayManifestModal
-            projectId={projectId}
-            skillNameIdentifier={displaySkillManifest}
-            onDismiss={() => dismissManifestModal(projectId)}
+            projectId={displaySkillManifestOnProjectId}
```
Comment (srinaath) [inline Composer/packages/client/src/pages/design/DesignPage.tsx:891]: why do we need a selector here. Cant it be just `projectId={skillId ?? projectId}` instead `projectId={displaySkillManifestOnProjectId}`

Comment (srinaath) [inline Composer/packages/client/src/pages/design/DesignPage.tsx:891]: why do we need a selector here. Cant it be just `projectId={skillId ?? projectId}` instead `projectId={displaySkillManifestOnProjectId}`

Comment (zhixzhan) [reply Composer/packages/client/src/pages/design/DesignPage.tsx:891]: This `displaySkillManifestOnProjectId` is from click project tree context menu, and skillId is from url
/bot/<botId>/skill/<skillId>

and when current workspace have multiple bot, if any of it request `create/edit manifest`, the modal should be display. that's what this selector do.

I thought about use local state just like in this page other `exportSkillModalInfo` `triggerModalInfo`, but the project tree click is not the only entrance of display it, so it's better to keep a global state. then it need a selector.

**Review by boydc2014 — DISMISSED** [review]

**Comment by boydc2014** [comment]
@zhixzhan looks like one last UT is failing

**Comment by srinaath** [comment]
@zhixzhan These links still dont work for me

**Comment by srinaath** [comment]
We have an issue with the hover interaction still.

https://microsoft.sharepoint.com/:i:/t/ConversationalAI785/Ef-TO6oI-lxAliy1Dv7unCEBRlRi8fWdJJQgkhT9OwhohQ?e=715ebh

**Comment by srinaath** [comment]
I think these tickets were supposed to be wrapped in this PR. 

#4920 #4919

File: Composer/packages/client/src/components/Modal/DisplayManifestModal.tsx [hunk 3]
```diff
@@ -16,7 +16,7 @@ import { IDialogContentStyles } from 'office-ui-fabric-react/lib/Dialog';
 import { IModalStyles } from 'office-ui-fabric-react/lib/Modal';
```
Comment (srinaath) [inline Composer/packages/client/src/components/Modal/DisplayManifestModal.tsx:16]: Requesting changes

**Review by srinaath — CHANGES_REQUESTED** [review]

**Comment by lei9444** [comment]
> I think these tickets were supposed to be wrapped in this PR. 
> 
> #4920 #4919

fixed this two

**Comment by lei9444** [comment]
> We have an issue with the hover interaction still.
> 
> https://microsoft.sharepoint.com/:i:/t/ConversationalAI785/Ef-TO6oI-lxAliy1Dv7unCEBRlRi8fWdJJQgkhT9OwhohQ?e=715ebh

fixed this one

**Comment by lei9444** [comment]
> 
> 
> @zhixzhan These links still dont work for me

fixed this one

**Comment by lei9444** [comment]
> We have an issue with the hover interaction still.
> 
> https://microsoft.sharepoint.com/:i:/t/ConversationalAI785/Ef-TO6oI-lxAliy1Dv7unCEBRlRi8fWdJJQgkhT9OwhohQ?e=715ebh

looks like the target(the icon) is too small, when move away the icon, the tooltip will dismiss

**Comment by lei9444** [comment]
use callout to replace the tooltip
callout

**Review by cwhitten — APPROVED** [review]

**Pull request merged**